npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/yarn_lockfile_updater.rb

```diff
@@ -56,6 +56,8 @@ def sub_dependencies
         def updated_yarn_lock(yarn_lock)
           base_dir = dependency_files.first.directory
           SharedHelpers.in_a_temporary_repo_directory(base_dir, repo_contents_path) do
+            # Pull any yarn data or plugins which may be stored with Git LFS
+            SharedHelpers.run_shell_command("git lfs pull --include .yarn")
```

Comment:
We wouldn't want to perform this step here, as this gets called for every dependency that gets updated, which would result in a lot of unnecessary work.

If we wanted to do this we'd have to do it here: https://github.com/dependabot/dependabot-core/blob/86b4fdf9054552c0ffc4b8ff145f0e65cf4a7473/common/lib/dependabot/file_fetchers/base.rb#L507, or override that method for the `Dependabot::NpmAndYarn::FileFetcher`.

We would probably want to inspect the yarn config in order to determine what path to pull, as the cache and plugin folders are configurable I believe

Reply:
I've updated to override `_clone_repo_contents` instead of doing this inside of `updated_yarn_lock`. See below for a question about the yarn config.